client/me/reauth-required/security-key-form.jsx

```diff
@@ -0,0 +1,109 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { Card } from '@automattic/components';
+import FormButton from 'components/forms/form-button';
+import { localize } from 'i18n-calypso';
+import Spinner from 'components/spinner';
+import { recordTracksEventWithClientId as recordTracksEvent } from 'state/analytics/actions';
+
+/**
+ * Style dependencies
+ */
+import './security-key-form.scss';
+import FormInputValidation from 'components/forms/form-input-validation';
+
+class SecurityKeyForm extends Component {
+	static propTypes = {
+		loginUserWithSecurityKey: PropTypes.func.isRequired,
+		onComplete: PropTypes.func,
+		recordTracksEvent: PropTypes.func.isRequired,
```

Comment:
We generally differentiate between the three types of props:

* Expected props - props that are passed to the component when it's rendered somewhere.
* Connected props - props that are returned from the `connect` call, generally returned in `mapStateToProps` or `mapDispatchToProps`
* High-order component props - props that are being provided by high order components.

For example, we would often do this:

```
static propTypes = {
	expectedProp: PropTypes.string,

	// Connected props
	foo: PropTypes.object,
	bar: PropTypes.string,

	// From localize() HoC
	translate: PropTypes.func.isRequired,

	// From withLocalizedMoment() HoC
	moment: PropTypes.func.isRequired,
};
```

Reply:
Addressed this and the other comments below in c1184303d09f26a3ed772567df78b4251b6cdb7c